gix/tests/object/tree/diff.rs

```diff
@@ -378,6 +379,84 @@ mod track_rewrites {
         Ok(())
     }
 
+    #[test]
+    fn jj_realistic_needs_to_be_more_clever() -> crate::Result {
```

Comment:
Due to the complexity of setting up the diffing in the tracker, I decided to add the test on `gix` level instead of on plumbing level, where everything is far more 'simulated'.

The benefit is that here one can define the final result exactly as one would want to see it, but with the downside that the implementation actually happens elsewhere, i.e. in the tracker, which has its own set of tests which should probably also be amended with boiled-down versions of this once the algorithm is adjusted.

One assume that that the algorithm changes will already change the outcome of the existing tests.